pkg/gadgets/internal/socketenricher/bpf/sockets-map.h

```diff
@@ -81,26 +81,38 @@ struct {
 static __always_inline struct sockets_value *
 gadget_socket_lookup(const struct __sk_buff *skb)
 {
-	// Only IPv4 is supported for now
-	if (load_half(skb, offsetof(struct ethhdr, h_proto)) != ETH_P_IP)
-		return 0;
-
 	struct sockets_key key = {0,};
+	int l4_off;
+
 	key.netns = current_netns;
-	key.family = AF_INET;
-	key.proto = load_byte(skb, ETH_HLEN + offsetof(struct iphdr, protocol));
+	key.family = load_half(skb, offsetof(struct ethhdr, h_proto));
+	switch (key.family) {
+	case ETH_P_IP:
+		key.family = AF_INET;
+		key.proto = load_byte(skb, ETH_HLEN + offsetof(struct iphdr, protocol));
+		l4_off = ETH_HLEN + sizeof(struct iphdr);
```

Comment:
It's missing the options field. I think we can use the [IHL (Internet Header Length)](https://en.wikipedia.org/wiki/Internet_Protocol_version_4#IHL), however we need to check that the verifier is happy.

Reply:
We have the sni gadget that parses the IHL:
https://github.com/inspektor-gadget/inspektor-gadget/blob/427603a57dff2d265f10d9a864fc093427dd12de/pkg/gadgets/trace/sni/tracer/bpf/snisnoop.c#L142-L146

I can try to do something similar.